### DISCUSSION TITLE
how to restore forked repository?

### DISCUSSION THREAD

**Discussion by MahjabeenTahir** [community/community#60775] [New to GitHub] [answered]
Hi, I forked a repository from the collaborator and made lots of changes but when I unfollow the collaborator git hub automatically deleted my forked repository. How to restore it please help me.

**Comment by yaxan** [comment, marked as answer]
A forked repository should remain there regardless of whether you unfollow because it creates a separate repo on your profile. If that is what happened though, there is no straightforward way to restore a deleted repo, so your best bet is to contact GitHub support.

For the forked repo, if you saved the changes locally, you can just reupload it to GitHub.

**Reply by MahjabeenTahir** [reply]
I cannot see that repo, how to reupload it?

**Reply by MahjabeenTahir** [reply]
I contacted github support and they forwarded me this link to ask about it in the community.

**Reply by yaxan** [reply]
If you have your forked repo on your computer (locally), you need to create a new repository on GitHub and push your local repository to that remote repository. You'd navigate in your command line to the local repo and run:
`git init
git add .
git commit -m "First commit"
git remote add origin <REPO URL>
git push -u origin main
`

**Reply by MahjabeenTahir** [reply]
No ..I dont understand.. where to write this command after creating a new repo

**Reply by MahjabeenTahir** [reply]
"Your old repository's clone URL"..... I dont know this...what to write?

**Reply by yaxan** [reply]
In your computer's terminal or command line, IF you have a local copy, meaning not on github but on your computer's hard drive. You go to github and create an empty repository, get the link to it from there, then you navigate in your command line to where the local copy of your repo is. from there you push your local repo to your remote repo (computer folder -> github repo). If this does not make sense to you, then I imagine your fork of the repo contained no changes from the original repository, and you should go in your browser history, go to the original author's repo, and fork it again. I recommend learning a bit about how git works, and practicing a bit with github, it will save you a lot of trouble.

**Reply by MahjabeenTahir** [reply]
It's not on my computer's hard drive but it is available. Now this is written: "Your old project requires credentials for read-only access. We will only temporarily store them for importing. When importing from Azure DevOps, please use a personal access token with read access to the repository being imported". Asking for login and Private access token.  I don't have.

**Reply by MahjabeenTahir** [reply]
I still cannot access it.....
